pkg/http/server.go

```diff
@@ -47,10 +49,11 @@ func NewServer(api API, config ServerConfig, tracingSpec config.TracingSpec, pip
 // StartNonBlocking starts a new server in a goroutine
 func (s *server) StartNonBlocking() {
 	handler :=
-		s.useProxy(
-			s.useCors(
-				s.useComponents(
-					s.useRouter())))
+		useAPIAuthentication(
```

Comment:
s. ?

Reply:
+ 1.  why not s.useAPIAuthentication( ...) ?

Reply:
it doesn't need anything from the server itself, for testability also its better to keep it without a struct ref.